controllers/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -155,6 +159,48 @@ func (f *admissionControllerFeature) Configure(dda *v2alpha1.DatadogAgent) (reqC
 			} else if ok && componentOverride.Image != nil {
 				f.agentSidecarConfig.imageTag = componentOverride.Image.Tag
 			}
+
+			// check selector slices If they have values.
+			for _, selector := range sidecarConfig.Selectors {
+				newSelector := &v2alpha1.Selector{}
+
+				if selector.NamespaceSelector != nil {
+					nsSelector := &metav1.LabelSelector{
+						MatchLabels:      selector.NamespaceSelector.MatchLabels,
+						MatchExpressions: selector.NamespaceSelector.MatchExpressions,
+					}
+					if len(nsSelector.MatchLabels) > 0 || len(nsSelector.MatchExpressions) > 0 {
+						newSelector.NamespaceSelector = nsSelector
+					}
```

Comment:
Could we instead consolidate this?
```suggestion
				if selector.NamespaceSelector != nil {
					newSelector.NamespaceSelector := &metav1.LabelSelector{
						MatchLabels:      selector.NamespaceSelector.MatchLabels,
						MatchExpressions: selector.NamespaceSelector.MatchExpressions,
					}
```